core/blockchain.go

```diff
@@ -1987,6 +1919,87 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 	return it.index, err
 }
 
+// blockProcessingResult is a summary of block processing
+// used for updating the stats.
+type blockProcessingResult struct {
+	usedGas  uint64
+	procTime time.Duration
+	status   WriteStatus
+}
+
+// processBlock executes and validates the given block. If there was no error
+// it writes the block and associated state to database.
+func (bc *BlockChain) processBlock(block *types.Block, statedb *state.StateDB, followupInterrupt atomic.Bool, start time.Time, setHead bool) (blockEndErr error, _ bool, _ *blockProcessingResult) {
```

Comment:
```golang
func flipper(x atomic.Bool) {
	x.Store(true)
}

func TestThing(t *testing.T) {
	var x atomic.Bool
	flipper(x)
	t.Logf("value is: %v", x.Load())
}
```
Outputs `value is: false`. 

TLDR; you cannot pass an `atomic.Bool` by value like this

Reply:
And actually, it surprises me that this is not flagged somewhere -- the atomic.Bool struct has a `_ noCopy` field which I thought would somehow trigger a compilation error or something like that